address-space-controller/src/main/java/io/enmasse/controller/TemplateInfraResourceFactory.java

```diff
@@ -96,10 +97,16 @@ private void prepareParameters(AddressSpace addressSpace,
         }
         parameters.put(TemplateParameter.MESSAGING_SECRET, serviceCertMapping.get("messaging").getSecretName());
 
-        String brokerSupportUser =  String.format("broker-support-%s", UUID.randomUUID());
-        String brokerSupportPwd = UUID.randomUUID().toString();
-        parameters.put(TemplateParameter.BROKER_SUPPORT_USER, Base64.getEncoder().encodeToString(brokerSupportUser.getBytes()));
-        parameters.put(TemplateParameter.BROKER_SUPPORT_PWD, Base64.getEncoder().encodeToString(brokerSupportPwd.getBytes()));
+        Optional<Secret> secret = kubernetes.getSecret("broker-support-" + infraUuid);
+        if (!secret.isPresent()) {
+            String brokerSupportUser =  String.format("broker-support-%s", UUID.randomUUID());
+            String brokerSupportPwd = UUID.randomUUID().toString();
+            parameters.put(TemplateParameter.BROKER_SUPPORT_USER, Base64.getEncoder().encodeToString(brokerSupportUser.getBytes()));
+            parameters.put(TemplateParameter.BROKER_SUPPORT_PWD, Base64.getEncoder().encodeToString(brokerSupportPwd.getBytes()));
+        } else {
+            parameters.put(TemplateParameter.BROKER_SUPPORT_USER, secret.get().getData().get("username"));
```

Comment:
@robshelly what if the secret is present but username and/or password is absent?  I think the missing ones should be recreated.

Reply:
lgtm